src/Model.php

```diff
@@ -538,18 +538,25 @@ public function fieldFactory($seed = [])
      */
     public function addFields($fields = [], $defaults = [])
     {
-        foreach ($fields as $field) {
-            if (is_string($field)) {
-                $this->addField($field, $defaults);
+        foreach ($fields as $key => $field) {
+            if (!is_int($key)) {
+                // field name can be passed as array key
+                $name = $key;
+                $seed = array_merge($defaults, is_array($field) ? $field : [$field]);
```

Comment:
Maybe use the shorthand format?
`$seed = array_merge($defaults, (array) $field);`

Reply:
sounds good, will update